specs/slider/slider.md

```diff
@@ -0,0 +1,201 @@
+A developer can inject labels on the slider as children, if they use <slider-label> elements those will be styled a specific way internally and the dev can override any part of that style they want. Alternatively the dev can put whatever they want in children and style as they wish.
+
+- *A designer customizing the component.*
+A designer can override any internal styling applied to the slotted labels, the background track, the slider track and the thumb(s) if appropriate.
+  
+### Features
+- Implement your own lower and upper thumb UI, default provided.
+- Implement your own background and slider track UI, default provided.
+- Developers can hook up to the `slider`'s `onValueChange` callback to be notified when the value has changed.
+- Customize the tick marks along the track.
+- Form association, if the `slider` component is part of a form the results will be submitted along with the form.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/slider)
+- [Material UI](https://material-ui.com/components/slider/)
+
+---
+
+### API
+
+*Component Name*:
+- `fast-slider`
+
+*Props/Attrs*
+- `min` - minimum allowed value for the slider
+- `max` - maximum allowed value for the slider
+- `step` - limits the values of the `slider` to increments of the `step` value added to the minimum value of the 
+`slider`'s total range.  The default value is 1. The minimum and maximum values of a `slider`'s range are always valid results regardless of the `step` prop. The `step` prop is used as the value for incrementing the thumb by pressing the arrow keys.
+- `mode`
+The slider component can be set to different modes using the `mode` prop.  The default is `single-value`.  The modes are as follows:
+ > - `single-value` -  A single thumb, no selected range is displayed.
+ > - `adjust-upper-value` -  A single thumb associated with the upper value of slider's selected range.  The *slider*'s front bar element displays and is anchored to the low end of the slider's current range as the user adjusts the maximum value.
+ > - `adjust-lower-value` -  A single thumb associated with the lower value of slider's selected range.  The *slider*'s front bar element displays and is anchored to the high end of the slider's current range as the user adjusts the minimum value.
+ > - `adjust-both` - in which case the selected range starts at 40% to 60% of the total range
+
+ - `value` - Allows authors to specify the initial selected range of the `slider`.  It defaults to a (step constrained) value at the midpoint on the `slider`'s total range for all modes except `adjust-both`. 
+
+*Events*
+- `change` - raise the change event for external parties to be informed of the value change.
+
+*Methods*
+- `onValueChanged` - handling the internal framework call when the value attribute is changed.
+
+*Slots*
+- `background-track`
+- `foreground-track`
+- `slider-track`

```

Comment:
Can you add details regarding how these are each different?

Reply:
I simplified the interface to handle just the default case. I'm pretty sure it's the only case we were using seeing as the mode enum wasn't exported outside of fast. Added descriptions to what remains